src/Microsoft.AspNet.FileProviders.Combined/CombinedDirectoryContents.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.FileProviders
+{
+    internal class CombinedDirectoryContents : IDirectoryContents
+    {
+        private readonly Lazy<List<IDirectoryContents>> _directoriesContents;
+        private readonly Lazy<List<IFileInfo>> _files;
+        private readonly Lazy<bool> _exists;
+
+        public CombinedDirectoryContents(IFileProvider[] fileProviders, string subpath)
+        {
+            _directoriesContents =new Lazy<List<IDirectoryContents>>(() =>
```

Comment:
Could we possibly just do this on a first chance basis rather than use `Lazy`?  i.e. in 

```C#
public IEnumerator<IFileInfo> GetEnumerator()
{
    EnsureInitialized();
    return _files;
}

public bool Exists
{
   get
   {
      EnsureInitialized();
      return _exists;
  }
}

private void EnsureInitialized()
{
    if (_directoryContents == null)
    {
           // Do the stuff from the constructor here
     }
}

```

We aren't dealing with concurrency here, so just deferring evaluation should suffice.